### PR TITLE
add run and exec commands

### LINKED ISSUE #128: Add 'exec' and 'run' commands similar to nvm
The node version manager (nvm) has two commands that might be useful in KVM: `exec` and `run`. We should add similar commands:

* dnvm run 1.0.0-beta2 <args> -> Invokes `dnx <args>` for using the 1.0.0-beta2 `dnx`
* dnvm exec 1.0.0-beta2 <cmd> -> Creates a new sub-shell with the 1.0.0-beta2 DNX on the PATH and runs `<cmd>` in it.

Both should take aliases in place of versions as well.

### PR DESCRIPTION
**Pull request by anurse**
Fixes #128 

/cc @glennc @davidfowl 

I believe the PowerShell tests don't pass yet but am not on a Windows box. I'll ensure they pass before merging, but it shouldn't affect the implementation or general structure of the tests (and the sh tests/impl all work) so I figured I'd start the review now.

### CODE REVIEW

**Comment by davidfowl** [comment]
AppVeryor says no for some reason

**Comment by anurse** [comment]
Well, it says no because the tests are broken, as I expected. I'll fix them up, but it's purely test issues, not implementation issues.

**Comment by glennc** [comment]
That's what everyone says. Right before the test reveals an implementation issue...

File: src/dnvm.ps1 [hunk 1]
```diff
@@ -1180,6 +1180,80 @@ function dnvm-use {
 
 <#
 .SYNOPSIS
+    Locates the dnx.exe for the specified version or alias and executes it, providing the remaining arguments
```
Comment (glennc) [inline src/dnvm.ps1]: Seems unfinished. "Locates the dnx.exe for the specified version or alias and executes it, providing the remaining arguments to dnx.exe"?

File: src/dnvm.ps1 [hunk 2]
```diff
@@ -1180,6 +1180,80 @@ function dnvm-use {
...
+    The arguments to pass to dnx.exe
+#>
+function dnvm-run {
+    param(
+        [Parameter(Mandatory=$false, Position=0)]
+        [string]$VersionOrAlias,
+        [Parameter(Mandatory=$false, Position=1, ValueFromRemainingArguments=$true)]
+        [object[]]$DnxArguments)
+
+    if([String]::IsNullOrWhiteSpace($VersionOrAlias)) {
```
Comment (glennc) [inline src/dnvm.ps1]: Doesn't this mean it is mandatory? Contrary to what the param says...

**Comment by glennc** [comment]
:shipit: Once you get the tests and such working :)

File: src/dnvm.ps1 (hunk 2 above)
Comment (anurse) [inline src/dnvm.ps1]: Yeah, we do that elsewhere though. If it's marked as mandatory, we get a not-so-pretty error message and prompt. This is to make the error a little clearer. I'll take a look at it though :)

File: src/dnvm.ps1 (hunk 1 above)
Comment (anurse) [inline src/dnvm.ps1]: Yep, will fix.

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**